weave/scorers/trust_scorer.py

```diff
@@ -0,0 +1,376 @@
+                )
+            elif scorer_cls == WeaveHallucinationScorerV1:
+                scorer_params["threshold"] = HALLUCINATION_SCORER_THRESHOLD
+                scorer_params["model_name_or_path"] = (
+                    self.hallucination_model_name_or_path
+                )
+            elif scorer_cls == WeaveToxicityScorerV1:
+                scorer_params["total_threshold"] = TOXICITY_TOTAL_THRESHOLD
+                scorer_params["category_threshold"] = TOXICITY_CATEGORY_THRESHOLD
+                scorer_params["model_name_or_path"] = self.toxicity_model_name_or_path
+            elif scorer_cls == WeaveFluencyScorerV1:
+                scorer_params["threshold"] = FLUENCY_SCORER_THRESHOLD
+                scorer_params["model_name_or_path"] = self.fluency_model_name_or_path
+            elif scorer_cls == WeaveCoherenceScorerV1:
+                scorer_params["model_name_or_path"] = self.coherence_model_name_or_path
+
+            # Initialize and store scorer
+            self._loaded_scorers[scorer_cls.__name__] = scorer_cls(**scorer_params)
+
+    def _preprocess_text(self, text: str) -> str:
+        """Preprocess text by handling emojis and length."""
+        if not text:
+            return text
+
+        # Replace emojis with their text representation while preserving spacing
+        text = self._emoji_pattern.sub(lambda m: f" {m.group(0)} ", text)
+
+        # Clean up multiple spaces and normalize whitespace
+        text = " ".join(text.split())
+
+        # Ensure proper sentence spacing
+        text = (
+            text.replace(" .", ".")
+            .replace(" ,", ",")
+            .replace(" !", "!")
+            .replace(" ?", "?")
+        )
+
+        return text
+
+    def _filter_inputs_for_scorer(
+        self, scorer: weave.Scorer, inputs: dict[str, Any]
+    ) -> dict[str, Any]:
+        """Filter inputs to match scorer's signature."""
+        scorer_params = signature(scorer.score).parameters
+        return {k: v for k, v in inputs.items() if k in scorer_params}
+
+    def _score_all(
+        self,
+        output: str,
+        context: Optional[Union[str, list[str]]] = None,
+        query: Optional[str] = None,
+    ) -> dict[str, Any]:
+        """Run all applicable scorers and return their raw results."""
+        # Preprocess inputs
+        processed_output = self._preprocess_text(output)
+        processed_context = (
+            self._preprocess_text(context) if isinstance(context, str) else context
+        )
+        processed_query = self._preprocess_text(query) if query else None
+
+        inputs: dict[str, Any] = {"output": processed_output}
+        if processed_context is not None:
+            inputs["context"] = processed_context
+        if processed_query is not None:
+            inputs["query"] = processed_query
+
+        results = {}
+
+        if self.run_in_parallel:
+            with ThreadPoolExecutor() as executor:
```

Comment:
You will want to add a test in the Evaluation framework for this flag.  In there, we also have thread parallelism, and it's not clear how these two will interact

Reply:
Basically just add the trust scorer as a scorer and run the eval framework across ~100 rows

Reply:
This scorer is slow, do we want to really do this?

Reply:
Well you need to test the parallelism somehow.  It's not obvious to me that this is safe

Reply:
we can turn off parallelism by default for the time being

Reply:
no, if you include parallelism then you need to test it.  Otherwise you have to remove it

Reply:
added a test